tests/models/test_google.py

```diff
@@ -1393,3 +1394,14 @@ class CountryLanguage(BaseModel):
             ),
         ]
     )
+
+
+async def test_google_model_usage_limit_exceeded(allow_model_requests: None, google_provider: GoogleProvider):
+    model = GoogleModel('gemini-2.5-flash', provider=google_provider)
+    agent = Agent(model=model)
+
+    with pytest.raises(UsageLimitExceeded, match='Exceeded the request_tokens_limit of 9 \\(request_tokens=12\\)'):
+        await agent.run(
```

Comment:
I'd love to verify somehow that this error is coming from the before-request check rather than the after-request check. Maybe we can do something like in `test_google_model_iter_stream` where we use `agent.iter` to collect all the events that were seen, and verify the list of events is empty.